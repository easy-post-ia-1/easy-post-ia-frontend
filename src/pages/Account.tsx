import { createUserAdapter } from '@adapters/user.adapter';
import AuthenticatedNavbar from '@components/navbar/AuthenticatedNavbar';
import { useGetAccount } from '@hooks/queries/user/useProfileQuery';
import {
  Avatar,
  Box,
  Container,
  Typography,
  Skeleton,
  Checkbox,
  FormControlLabel,
  Alert,
  Grid,
  Paper,
  useMediaQuery,
  useTheme,
  Button,
} from '@mui/material';
import { useCompanySocialStatus } from '@hooks/queries/user/useCompanySocialStatusQuery';
import BottomNavigationMobile from '@components/navbar/BottomNavigationMobile';
import { useTranslation } from 'react-i18next';
import { capitalizeFirst } from '@utils/helpers';
import { useNavigate } from 'react-router-dom';
import { useUserStore } from '@stores/userStore';
import { useAuthStore } from '@stores/useAuthStore';

function Account() {
  const { t } = useTranslation();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('md'));
  const navigate = useNavigate();
  const { logout } = useUserStore();
  const { updateErrorToken, updateToken } = useAuthStore();
  const { data = { user: {} } } = useGetAccount();
  const user = createUserAdapter(data?.user);
  const {
    data: socialStatusData,
    isLoading: isLoadingSocialStatus,
    isError: isErrorSocialStatus,
    error: errorSocialStatus,
  } = useCompanySocialStatus();

  const handleLogout = () => {
    logout();
    updateErrorToken(null);
    updateToken(null);
    navigate('/login');
  };

  return (
    <>
      <AuthenticatedNavbar />
      <Container maxWidth="sm" sx={{ mt: 4, pb: isMobile ? 8 : 0 }}>
        {/* Existing User Profile Box */}
        <Box display="flex" flexDirection="column" alignItems="center">
          <Avatar alt="Profile Image" sx={{ width: 100, height: 100, mb: 2 }} />
          <Typography variant="h6">{user?.username}</Typography>
          <Typography variant="body1" color="textSecondary">
            {user?.email}
          </Typography>
          <Typography variant="body2" color="textSecondary" sx={{ mt: 0.5 }}>
            {capitalizeFirst(user?.role)}
          </Typography>
        </Box>

        {/* New Social Network Section Box */}
        <Box sx={{ mt: 4 }}>
          <Typography variant="h5" gutterBottom component="div">
            {t('account.socialStatus.title')}
          </Typography>
          {isLoadingSocialStatus && (
            <>
              <Skeleton variant="text" width="40%" height={40} />
              <Skeleton variant="rectangular" width="100%" height={30} sx={{ mt: 1, mb: 1 }} />
              <Skeleton variant="rectangular" width="100%" height={30} />
            </>
          )}
          {isErrorSocialStatus && (
            <Alert severity="error" sx={{ mt: 2 }}>
              {errorSocialStatus?.message || t('account.socialStatus.errorLoad')}
            </Alert>
          )}
          {!isLoadingSocialStatus && !isErrorSocialStatus && (
            <>
              {socialStatusData?.networks && socialStatusData.networks.length > 0 ? (
                <Paper elevation={1} sx={{ p: 2, mt: 2 }}>
                  <Grid container spacing={1}>
                    {socialStatusData.networks.map((network) => (
                      <Grid item xs={12} key={network.name}>
                        <FormControlLabel
                          control={<Checkbox checked={network.hasCredentials} disabled />}
                          label={network.name}
                        />
                      </Grid>
                    ))}
                  </Grid>
                </Paper>
              ) : (
                <Paper elevation={1} sx={{ p: 2, mt: 2 }}>
                  <Typography>
                    {socialStatusData ? t('account.socialStatus.noAccountsConfigured') : t('account.socialStatus.unavailable')}
                  </Typography>
                </Paper>
              )}
            </>
          )}
        </Box>

        {/* Logout Button - Mobile Only */}
        {isMobile && (
          <Box sx={{ mt: 4, display: 'flex', justifyContent: 'center' }}>
            <Button
              variant="outlined"
              color="error"
              onClick={handleLogout}
              sx={{ minWidth: '200px' }}
            >
              {t('account.logout')}
            </Button>
          </Box>
        )}
      </Container>
      <BottomNavigationMobile />
    </>
  );
}

export default Account;
