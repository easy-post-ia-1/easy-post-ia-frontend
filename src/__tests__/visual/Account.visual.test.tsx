import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { I18nextProvider } from 'react-i18next';
import i18n from '../../_i18n/index';
import Account from '../../pages/Account';

// Mock react-i18next
vi.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock t function to return the key itself
  }),
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  },
  I18nextProvider: ({ children }: { children: React.ReactNode }) => children,
}));

// Mock AuthenticatedNavbar
vi.mock('@components/navbar/AuthenticatedNavbar', () => ({
  default: () => <div data-testid="mocked-navbar">Mocked Navbar</div>,
}));

// Mock BottomNavigationMobile
vi.mock('@components/navbar/BottomNavigationMobile', () => ({
  default: () => <div data-testid="mocked-bottom-nav">Mocked Bottom Nav</div>,
}));

// Mock useUserStore
vi.mock('@stores/userStore', () => ({
  useUserStore: () => ({
    logout: vi.fn(),
  }),
}));

// Mock useAuthStore
vi.mock('@stores/useAuthStore', () => ({
  useAuthStore: () => ({
    updateErrorToken: vi.fn(),
    updateToken: vi.fn(),
  }),
}));

// Mock capitalizeFirst helper
vi.mock('@utils/helpers', () => ({
  capitalizeFirst: (str: string) => str ? str.charAt(0).toUpperCase() + str.slice(1) : '',
}));

// Mock useGetAccount hook
vi.mock('@hooks/queries/user/useProfileQuery', () => ({
  useGetAccount: () => ({
    data: { user: { id: 1, username: 'visual', email: 'visual@mail.com' } },
    isLoading: false,
    isError: false,
  }),
}));

// Mock useCompanySocialStatus hook
vi.mock('@hooks/queries/user/useCompanySocialStatusQuery', () => ({
  useCompanySocialStatus: () => ({
    data: { networks: [] },
    isLoading: false,
    isError: false,
    error: null,
  }),
}));

// Mock createUserAdapter
vi.mock('@adapters/user.adapter', () => ({
  createUserAdapter: (user: any) => ({
    ...user,
    role: 'employer',
  }),
}));

describe('Visual: Account snapshot', () => {
  let queryClient: QueryClient;

  beforeEach(() => {
    queryClient = new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
        },
      },
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <QueryClientProvider client={queryClient}>
          <I18nextProvider i18n={i18n}>
            <Account />
          </I18nextProvider>
        </QueryClientProvider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 