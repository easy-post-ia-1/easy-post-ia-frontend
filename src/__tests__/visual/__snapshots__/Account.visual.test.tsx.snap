// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Visual: Account snapshot > matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="mocked-navbar"
  >
    Mocked Navbar
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-14owtho-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-dvxtzn"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-165ewxr-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-1mo2pzk-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
      >
        visual
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1ro85z9-MuiTypography-root"
      >
        visual@mail.com
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-14mf82g-MuiTypography-root"
      >
        Employer
      </p>
    </div>
    <div
      class="MuiBox-root css-h5fkc8"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1f9xtph-MuiTypography-root"
      >
        account.socialStatus.title
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1blig1b-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          account.socialStatus.noAccountsConfigured
        </p>
      </div>
    </div>
  </div>
  <div
    data-testid="mocked-bottom-nav"
  >
    Mocked Bottom Nav
  </div>
</DocumentFragment>
`;
